File: data-ingestion/src/data_ingestion/app/api/api_v2/endpoints/tickers/etf.py
import pandas as pd
from fastapi import APIRouter

from ion_clients.services.postgres.postgres_service import _get_postgres_engine

from data_ingestion.app.api.api_v2.postgres.schemas.data.tickers.params import (
    ETFInfoParams,
    ETFInfosParams,
)

router = APIRouter(
    tags=["tickers", "etf"],
)


@router.get("/health")
def health_check():
    return {"status": "healthy"}


@router.post(f"/info")
def get_etf_info(
    params: ETFInfoParams,
):
    return


@router.post(f"/infos")
def get_etfs_info(
    params: ETFInfosParams,
):
    pass

File: frontend/src/endpoints/clients/etf.ts
import { ENDPOINTS } from 'endpoints/endpoints';
import { ETFDataSchema } from 'endpoints/schema/etf';
import { request } from 'services/request';

export const getETFInfo = (ticker: string) => {
    return request('data-ingestion').post<ETFDataSchema>(ENDPOINTS.PRIVATE.ETF_INFO, {
        ticker: ticker,
    });
};
